locust/main.py

```diff
@@ -241,12 +241,20 @@ def timelimit_stop():
     if options.headless:
         # headless mode
         if options.master:
-            # what for worker nodes to connect
+            # wait for worker nodes to connect
             while len(runner.clients.ready) < options.expect_workers:
                 logging.info("Waiting for workers to be ready, %s of %s connected",
                              len(runner.clients.ready), options.expect_workers)
                 time.sleep(1)
         if not options.worker:
+            # apply headless mode defaults
+            if options.num_clients is None:
+                options.num_clients = 1
+            if options.hatch_rate is None:
+                options.hatch_rate = 1
+            if options.step_clients is None:
+                options.step_clients = 1
```

Comment:
I think we should consider changing so that the `-c` and `-r` arguments are mandatory in `--headless` mode.

Reply:
Should I change this now or would you prefer to discuss it more?

Reply:
Hmm, let's keep the current behaviour for this PR. We can do a separate issue/PR for that later :).